### PR TITLE
test: migrate test_ignore

### PR DESCRIPTION
**Pull request by Suor**

### CODE REVIEW

**Comment by efiop** [comment]
@Suor Windows tests failed, please take a look.

**Comment by efiop** [comment]
@Suor Still failing 🙂

**Comment by Suor** [comment]
@efiop debugging against travis )

File: tests/func/test_ignore.py [hunk 1]
```diff
@@ -1,186 +1,121 @@
...
-        self.assertNotIn(forbidden_path, all_paths)
 
-    def test_ignore_in_child_dir_unicode(self):
-        ignore_file = os.path.join(self.dvc.root_dir, DvcIgnore.DVCIGNORE_FILE)
-        with open(ignore_file, "wb") as fobj:
-            fobj.write(cast_bytes(self.UNICODE, "utf-8"))
+def test_ignore_unicode(tmp_dir, dvc):
+    tmp_dir.gen({"dir": {"other": "text"}})
+    # Path() doesn't handle unicode paths in Windows/Python 2
+    # I don't know to debug it further, waiting till Python 2 EOL
```
Comment (mroutis) [inline tests/func/test_ignore.py]: :+1:

File: tests/func/test_ignore.py [hunk 2]
```diff
@@ -1,186 +1,121 @@
...
+        dvc.add("dir")
 
-def test_metadata_changed_on_non_ignored_file_deletion(dvc_repo, repo_dir):
-    mtime, size = get_mtime_and_size(repo_dir.DATA_DIR, dvc_repo.dvcignore)
 
-    os.remove(repo_dir.DATA_SUB)
-    new_mtime_sig, new_size = get_mtime_and_size(
-        repo_dir.DATA_DIR, dvc_repo.dvcignore
-    )
+# @efiop why do we need to parametrize this?
```
Comment (mroutis) [inline tests/func/test_ignore.py]: @efiop ^

Comment (efiop) [reply tests/func/test_ignore.py]: @Suor @mroutis because we need to ensure that subdirs work too. Or am I missing something?

Comment (Suor) [reply tests/func/test_ignore.py]: Testing that `.dvcignore` works for subdirs and testing that ignored `.dvcignore` files won't be collected looks like two different things. I am not strong on this though, so if you think this is ok, then resolve it.

Comment (efiop) [reply tests/func/test_ignore.py]: Yeah, could be better. Let's keep it for now though, please feel free to create an issue.

P.S. merged before removing this comment, will fix upstream.

**Pull request merged**